diff --git a/service_links.api.php b/service_links.api.php
index b28eb17..3097261 100644
--- a/service_links.api.php
+++ b/service_links.api.php
@@ -65,3 +65,27 @@ function hook_service_links_alter(&$links) {
     $links['myservice']['icon'] = 'http://drupal.org/misc/favicon.ico';
   }
 }
+
+/**
+ * Allows alteration of the service link settings for a given node.
+ *
+ * @param array $settings
+ *   Reference to the array of service link settings for a node.
+ * @param node
+ *   The fully loaded node object that the settings correspond to.
+ *
+ * @see drupal_alter();
+ * @see _service_links_get_tags();
+ */
+function hook_service_links_node_settings_alter(&$settings, $node) {
+  // Use a site-specific URL shortener.
+  $short_url = "http://example.com/$node->nid";
+  $settings['subst']['short-url'] = $short_url;
+  $settings['subst']['encoded-short-url'] = urlencode($short_url);
+  $settings['subst']['raw-encoded-short-url'] = urlencode($short_url);
+
+  // Hide a specific service on nodes of a specific type.
+  if ($node->type == 'notweets') {
+    unset($settings['link_show']['twitter']);
+  }
+}
diff --git a/service_links.module b/service_links.module
index fd71d25..da4ce56 100644
--- a/service_links.module
+++ b/service_links.module
@@ -60,6 +60,9 @@ function service_links_hook_info() {
     'service_links_alter' => array(
       'group' => 'service_links',
     ),
+    'service_links_node_settings_alter' => array(
+      'group' => 'service_links',
+    ),
   );
 
   return $hooks;
@@ -792,6 +795,8 @@ function _service_links_get_tags($node, &$settings) {
     'title' => $title,
     'front-page' => $front_page,
   );
+  // Invoke an alter hook to let other modules modify these values if needed.
+  drupal_alter('service_links_node_settings', $settings, $node);
 }
 
 /**
