diff --git a/template.php b/template.php
index 75cc386..02da299 100644
--- a/template.php
+++ b/template.php
@@ -107,7 +107,9 @@ function zen_preprocess_html(&$variables, $hook) {
   // Add variables and paths needed for HTML5 and responsive support.
   $variables['base_path'] = base_path();
   $variables['path_to_zen'] = drupal_get_path('theme', 'zen');
-  $html5_respond_meta = theme_get_setting('zen_html5_respond_meta');
+  // Get settings for HTML5 and responsive support. array_filter() removes
+  // items from the array that have been disabled.
+  $html5_respond_meta = array_filter((array) theme_get_setting('zen_html5_respond_meta'));
   $variables['add_respond_js']          = in_array('respond', $html5_respond_meta);
   $variables['add_html5_shim']          = in_array('html5', $html5_respond_meta);
   $variables['default_mobile_metatags'] = in_array('meta', $html5_respond_meta);
